### ISSUE TITLE
Could we get a CSV import option

### ISSUE THREAD

**Issue by alexlehm** [linkwarden/linkwarden#467]
**Is your feature request related to a problem? Please describe.**
I have links in the form of CSV (usually created via perl) that I would like to import in linkwarden, the same option is available as export on some bookmark tools

**Describe the solution you'd like**
A simple link list as CSV, maybe with a category and tags field (optional)

**Describe alternatives you've considered**
I am considering writing a perl script to convert from CSV to the netscape html format to make this possible without changing the program

**Additional context**
a similar issue is described with https://github.com/linkwarden/linkwarden/issues/335

**Comment by igorer88** [comment]
Hello!

There are no options for massive imports from file?

**Comment by alexlehm** [comment]
> Hello!
> 
> There are no options for massive imports from file?

Only when using Netscape export format

**Comment by xrogeman** [comment]
> > Hello!
> > There are no options for massive imports from file?
> 
> Only when using Netscape export format

could you give us more info about this?

**Comment by alexlehm** [comment]
> > > Hello!
> > > There are no options for massive imports from file?
> > 
> > 
> > Only when using Netscape export format
> 
> could you give us more info about this?

Linkwarden supports importing bookmark dumps in a HTML format that is usually called Netscape bookmark file, which other browsers and pages also support but with varying features, e.g. with or without timestamps, this makes it impossible to import some bookmark dumps, e.g. I tried Pocket and that didn't work.

I have written a small Perl script to convert the export from 2 tools to the format that Linkwarden can read directly, as a simple fix that might work, but I would prefer a more compatible import parser or one that supports CSV which is better over all I think.

**Comment by daniel31x13** [comment]
Thanks for the suggestion, but I'm closing this due to the limited demand. Sorry.

**Comment by prohtex** [comment]
> Thanks for the suggestion, but I'm closing this due to the limited demand. Sorry.

Hi @daniel31x13 thanks for all your hard work on this amazing app. I for one would love to see a txt file import for links!

**Comment by alexlehm** [comment]
@prohtex i have considered the question a bit further and I think it would be easy to make a txt to Netscape bookmarks file converter to enable txt and csv imports and simply run that externally before uploading the file. Maybe someone could give this a try, currently I don't have time, but I could at some point

**Comment by prohtex** [comment]
> [@prohtex](https://github.com/prohtex) i have considered the question a bit further and I think it would be easy to make a txt to Netscape bookmarks file converter to enable txt and csv imports and simply run that externally before uploading the file. Maybe someone could give this a try, currently I don't have time, but I could at some point

Yes, I figured it out when I realized "Netscape format" is just a list of href links

**Comment by mattes3** [comment]
Just discovered this: https://csv-to-bookmarks.glitch.me/
